### PR TITLE
Add Notes Regarding Input Validation Testing section

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add Notes Regarding Input Validation Testing section.

### Details

- Follow up #5571.

### CODE REVIEW

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -639,6 +639,81 @@ For example, the following writes all images generated by ``pytest`` to
...
+included in a single `.py` file. The test cases are all stored in
+``tests/core/typing/validation_cases``.
+
+The tests can be executed with:
+
+.. code:: python
+
+    pytest tests/core/typing
+
+When executed, a single instance of ``Mypy`` will statistically analyze all the
```
Comment (user27182) [inline CONTRIBUTING.rst:705]:
```suggestion
When executed, a single instance of ``Mypy`` will statically analyze all the
```

**Review by user27182 — CHANGES_REQUESTED** [review]

**Review by user27182 — APPROVED** [review]

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -639,6 +639,81 @@ For example, the following writes all images generated by ``pytest`` to
...
+
+The tests can be executed with:
+
+.. code:: python
+
+    pytest tests/core/typing
+
+When executed, a single instance of ``Mypy`` will statically analyze all the
+test cases. The actual revealed types by ``Mypy`` are compared against the
+``EXPECTED_TYPE``is defined by each test case.
```
Comment (user27182) [inline CONTRIBUTING.rst:707]:
```suggestion
``EXPECTED_TYPE`` is defined by each test case.
```

Comment (user27182) [reply CONTRIBUTING.rst:707]: This is causing a build failure.

**Review by user27182 — APPROVED** [review]

**Pull request merged**